src/components/hook/useStable.ts

```diff
@@ -0,0 +1,23 @@
+import { useRef, useEffect, useCallback } from 'react';
+
+// useEvent hook은 React 18.3 이상 (React Canary / experimental build에서만 사용 가능)
+// useStable 직접 구현: 최신 fn을 참조하면서도, 함수 참조는 변하지 않는 안전한 콜백 반환
+
+const useStable = <T extends (...args: any[]) => any>(fn: T): T => {
+  // 최신 함수 참조
+  const fnRef = useRef<T>(fn);
+
+  // 리렌더마다 최신 함수 갱신
+  useEffect(() => {
+    fnRef.current = fn;
+  }, [fn]);
```

Comment:
저도 이거 똑같이 한 번 만들어봤는데,
useEffect 시점에 최신화를 하게 되면 최초에 렌더링 되는 시점에 fn의 값이 달라지는..? 그런 문제가 있어요.

그래서 


```suggestion
  const fnRef = useRef<T>(fn);
  fnRef.current = fn;
```

이렇게 렌더링할 때 마다 최신화를 해줘야 하네요